src/lib/utils.js

```diff
@@ -102,3 +102,15 @@ export function saveBlob(blob, filename) {
   link.click();
   // URL.revokeObjectURL(url); breaks Firefox...
 }
+
+// Compares 2 vector objects up to size 4
+// Expect v1 and v2 to take format {x: number, y: number, z: number, w:number}
+// Smaller vectors (ie. vec2) should work as well since their z & w vals will be the same (undefined)
+export function areVectorsEqual(v1, v2) {
+  return (
+    Object.is(v1.x, v2.x) &&
```

Comment:
I would probably use the usual `v1.x === v2.x`

Reply:
No, that's the point of the fix here.
We use Object.is instead of === for comparison here so that comparing two NaN doesn't trigger an infinite update. Object.is(NaN, NaN) is true, NaN === NaN is false

Reply:
We added the above comment in NumberWidget where we also used Object.is. See the context here
https://github.com/aframevr/aframe-inspector/issues/677#issuecomment-1429366813